pkg/core/pipeline/scm/main.go

```diff
@@ -115,6 +119,63 @@ func (s *Scm) GenerateSCM() error {
 	return nil
 }
 
+// GetScmConfigFromDir returns the Scm configuration guessed from a directory on the filesystem, using the gitHandler passed as argument
+// An error is return if it fails to recognize the kind of SCM or if something goes wrong
+func GetScmConfigFromDir(workingDir string, gitHandler gitgeneric.GitHandler) (Config, error) {
+	remotes, err := gitHandler.RemoteURLs(workingDir)
+	if err != nil {
+		return Config{}, err
+	}
+	logrus.Debugf("Found the following remotes in %s: %s", workingDir, remotes)
+
+	originRemoteURL, ok := remotes["origin"]
```

Comment:
I am wondering if it's worthwhile to allow the configuration of originremote via the scm spec